api/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -42,12 +42,17 @@ type ForeignClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	ClusterID        string        `json:"clusterID"`
-	Namespace        string        `json:"namespace"`
-	Join             bool          `json:"join"`
-	ApiUrl           string        `json:"apiUrl"`
-	DiscoveryType    DiscoveryType `json:"discoveryType"`
-	AllowUntrustedCA bool          `json:"allowUntrustedCA"`
+	ClusterIdentity  *ClusterIdentity `json:"clusterIdentity"`
+	Namespace        string           `json:"namespace"`
+	Join             bool             `json:"join"`
```

Comment:
What does this mean?
`Join` looks meaningless to me. `AutoJoin`, maybe?
@palexster ?

Reply:
This is the flag to set to start the peering process, that can be set or because of an `AutoJoin` in Config or manually (or by dashboard) by the user

Reply:
I totally agree that "Join" does not explicit well the status of the peering, maybe something like "Peered". I suggest to handle this in a dedicated PR, since I feel it requires a wider refactoring which involve many components.